File: ViewModels/Pages/SensitivityMatchInfoBandViewModel.vb
Imports System.Windows
Imports WpfApp1.Infrastructure

Namespace ViewModels.Pages
    Public Class SensitivityMatchInfoBandViewModel
        Inherits BindableBase

        Private _labelText As String
        Private _deviceTitle As String
        Private _bindingStateText As String
        Private _metaText As String
        Private _bandHorizontalAlignment As HorizontalAlignment
        Private _bandScaleX As Double
        Private _gapAfterTitle As GridLength
        Private _gapAfterBindingState As GridLength

        Public Property LabelText As String
            Get
                Return _labelText
            End Get
            Set(value As String)
                SetProperty(_labelText, value)
            End Set
        End Property

        Public Property DeviceTitle As String
            Get
                Return _deviceTitle
            End Get
            Set(value As String)
                SetProperty(_deviceTitle, value)
            End Set
        End Property

        Public Property BindingStateText As String
            Get
                Return _bindingStateText
            End Get
            Set(value As String)
                If SetProperty(_bindingStateText, value) Then
                    RefreshLayoutMetrics()
                End If
            End Set
        End Property

        Public Property MetaText As String
            Get
                Return _metaText
            End Get
            Set(value As String)
                If SetProperty(_metaText, value) Then
                    RefreshLayoutMetrics()
                End If
            End Set
        End Property

        Public Property BandScaleX As Double
            Get
                Return _bandScaleX
            End Get
            Set(value As Double)
                If SetProperty(_bandScaleX, value) Then
                    RefreshLayoutMetrics()
                End If
            End Set
        End Property

        Public Property BandHorizontalAlignment As HorizontalAlignment
            Get
                Return _bandHorizontalAlignment
            End Get
            Set(value As HorizontalAlignment)
                SetProperty(_bandHorizontalAlignment, value)
            End Set
        End Property

        Public Property GapAfterTitle As GridLength
            Get
                Return _gapAfterTitle
            End Get
            Private Set(value As GridLength)
                SetProperty(_gapAfterTitle, value)
            End Set
        End Property

        Public Property GapAfterBindingState As GridLength
            Get
                Return _gapAfterBindingState
            End Get
            Private Set(value As GridLength)
                SetProperty(_gapAfterBindingState, value)
            End Set
        End Property

        Private Sub RefreshLayoutMetrics()
            Dim hasBindingState = Not String.IsNullOrWhiteSpace(_bindingStateText)
            Dim hasMeta = Not String.IsNullOrWhiteSpace(_metaText)

            GapAfterTitle = If(hasBindingState, New GridLength(16), New GridLength(0))
            GapAfterBindingState = If(hasMeta, New GridLength(16), New GridLength(0))
        End Sub
    End Class
End Namespace

File: src/ClickSyncMouseTester/ViewModels/Pages/SensitivityMatchInfoBandViewModel.cs
using ClickSyncMouseTester.Infrastructure;
using System.Windows;

namespace ClickSyncMouseTester.ViewModels.Pages;

public class SensitivityMatchInfoBandViewModel : BindableBase
{
    private string _labelText;

    private string _deviceTitle;

    private string _bindingStateText;

    private string _metaText;

    private HorizontalAlignment _bandHorizontalAlignment;

    private double _bandScaleX;

    private GridLength _gapAfterTitle;

    private GridLength _gapAfterBindingState;

    public string LabelText
    {
        get
        {
            return _labelText;
        }
        set
        {
            SetProperty(ref _labelText, value, "LabelText");
        }
    }

    public string DeviceTitle
    {
        get
        {
            return _deviceTitle;
        }
        set
        {
            SetProperty(ref _deviceTitle, value, "DeviceTitle");
        }
    }

    public string BindingStateText
    {
        get
        {
            return _bindingStateText;
        }
        set
        {
            if (SetProperty(ref _bindingStateText, value, "BindingStateText"))
            {
                RefreshLayoutMetrics();
            }
        }
    }

    public string MetaText
    {
        get
        {
            return _metaText;
        }
        set
        {
            if (SetProperty(ref _metaText, value, "MetaText"))
            {
                RefreshLayoutMetrics();
            }
        }
    }

    public double BandScaleX
    {
        get
        {
            return _bandScaleX;
        }
        set
        {
            if (SetProperty(ref _bandScaleX, value, "BandScaleX"))
            {
                RefreshLayoutMetrics();
            }
        }
    }

    public HorizontalAlignment BandHorizontalAlignment
    {
        get
        {
            return _bandHorizontalAlignment;
        }
        set
        {
            SetProperty(ref _bandHorizontalAlignment, value, "BandHorizontalAlignment");
        }
    }

    public GridLength GapAfterTitle
    {
        get
        {
            return _gapAfterTitle;
        }
        private set
        {
            SetProperty(ref _gapAfterTitle, value, "GapAfterTitle");
        }
    }

    public GridLength GapAfterBindingState
    {
        get
        {
            return _gapAfterBindingState;
        }
        private set
        {
            SetProperty(ref _gapAfterBindingState, value, "GapAfterBindingState");
        }
    }

    private void RefreshLayoutMetrics()
    {
        bool hasBindingStateText = !string.IsNullOrWhiteSpace(_bindingStateText);
        bool hasMetaText = !string.IsNullOrWhiteSpace(_metaText);
        GapAfterTitle = (hasBindingStateText ? new GridLength(16.0) : new GridLength(0.0));
        GapAfterBindingState = (hasMetaText ? new GridLength(16.0) : new GridLength(0.0));
    }
}





